packages/world/src/modules/uniqueentity/UniqueEntitySystem.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { System } from "../../System.sol";
+
+import { UniqueEntity } from "./tables/UniqueEntity.sol";
+
+import { NAMESPACE, TABLE_NAME } from "./constants.sol";
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+contract UniqueEntitySystem is System {
+  /**
+   * Increment and get an entity nonce.
+   */
+  function getUniqueEntity() public virtual returns (bytes32 uniqueEntity) {
+    uint256 tableId = uint256(ResourceSelector.from(NAMESPACE, TABLE_NAME));
+    uniqueEntity = bytes32(uint256(UniqueEntity.get(tableId)) + 1);
```

Comment:
does this guarantee uniqueness in any way?

wonder if we should keccak256 this with some other entropy (table ID maybe) to reduce chance of collisions?

Reply:
no, but it also shouldn't be an issue as entities aren't user-supplied, the only collision I can think of is SingletonId (which shouldn't be needed with singleton tables)